modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -1855,9 +1855,40 @@ public void testObjectFieldSettings() throws Exception {
 	@Test
 	public void testUpdateRequired() throws Exception {
 
-		// Deletion type cascade
+		// Non relationship object field
 
 		ObjectDefinition objectDefinition1 = _publishCustomObjectDefinition();
+
+		ObjectField objectField1 = _addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).required(
+				true
+			).build());
+
+		Assert.assertTrue(objectField1.isRequired());
+
+		objectField1.setRequired(false);
+
+		objectField1 = _addOrUpdateCustomObjectField(objectField1);
+
+		Assert.assertFalse(objectField1.isRequired());
+
+		objectField1.setRequired(true);
+
+		ObjectField finalObjectField = objectField1;
+
+		AssertUtils.assertFailure(
+			ObjectFieldRequiredException.class, null,
+			() -> _addOrUpdateCustomObjectField(finalObjectField));
```

Comment:
Since this code block relies on the `_addOrUpdateCustomObjectField()` method, this code should not be in this method test `testUpdateRequired()` because it only tests the method `_objectFieldLocalService.updateRequired()`.

Reply:
I think you should add this code in this method `testAddOrUpdateCustomObjectField()`.